plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoClientModule.java

```diff
@@ -46,24 +50,33 @@ public static MongoSession createMongoSession(TypeManager typeManager, MongoClie
     {
         requireNonNull(config, "config is null");
 
-        MongoClientOptions.Builder options = MongoClientOptions.builder();
-
-        options.connectionsPerHost(config.getConnectionsPerHost())
-                .connectTimeout(config.getConnectionTimeout())
-                .socketTimeout(config.getSocketTimeout())
-                .socketKeepAlive(config.getSocketKeepAlive())
-                .sslEnabled(config.getSslEnabled())
-                .maxWaitTime(config.getMaxWaitTime())
-                .maxConnectionIdleTime(config.getMaxConnectionIdleTime())
-                .minConnectionsPerHost(config.getMinConnectionsPerHost())
+        MongoClientSettings.Builder options = MongoClientSettings.builder();
+        options.writeConcern(config.getWriteConcern().getWriteConcern())
                 .readPreference(config.getReadPreference().getReadPreference())
-                .writeConcern(config.getWriteConcern().getWriteConcern());
+                .applyToConnectionPoolSettings(builder -> builder.maxConnectionIdleTime(config.getMaxConnectionIdleTime(), SECONDS)
```

Comment:
The property is milliseconds in MongoClientConfig. Please check subsequent time properties as well.

Reply:
nit: Move `.maxConnectionIdleTime...` to next line.